R/pca.R

```diff
@@ -72,3 +72,17 @@ cuda_ml_pca <- function(x,
 cuda_ml_inverse_transform.cuda_ml_pca <- function(model, x, ...) {
   .pca_inverse_transform(model = model, x = as.matrix(x))
 }
+
+cuda_ml_get_state.cuda_ml_pca <- function(model) {
+  model_state <- .pca_get_state(model)
+  class(model_state) <- c("cuda_ml_pca_model_state", class(model_state))
+
+  model_state
+}
+
+cuda_ml_set_state.cuda_ml_pca_model_state <- function(model_state) {
+  model <- .pca_set_state(model_state)
+  class(model) <- c("cuda_ml_pca", "cuda_ml_model", class(model))
```

Comment:
I think `class(model)` here returns `"list"`. Why do you want the model to inherit from `"list"`?

Reply:
No reason in particular. It's just that at the moment there is no need to "hide" the fact that this model object is a list.